### PR TITLE
SONIC-1554 BpkFlightLeg carrier logo correction

### PR DESCRIPTION
**Pull request by xo0ps**
[SONIC-1554](https://skyscanner.atlassian.net/browse/SONIC-1554)

This PR:
- adds `BpkBorderRadius.Xs = 4dp` radius to corners of `carrierLogoContent`
- adds `BpkTheme.colors.textOnDark = White in both Light and Dark modes` background color to `carrierLogoContent`
- adds `BpkSpacing.Lg = 24dp` as size to `carrierLogoContent`
- fixes long description and long stops info layout issue

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt [hunk 1]
```diff
@@ -74,7 +77,9 @@ fun BpkFlightLeg(
         Row(modifier = Modifier.fillMaxWidth()) {
             carrierLogoContent?.let {
                 Box(
-                    modifier = Modifier.padding(top = BpkSpacing.Sm),
+                    modifier = Modifier
+                        .padding(top = BpkSpacing.Sm)
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt:81]: Worth also adding the size as a constraint here

Comment (xo0ps) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt:81]: I thought about it, too. I'll add it.

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt [hunk 2]
```diff
@@ -73,13 +77,19 @@ fun BpkFlightLeg(
...
             carrierLogoContent?.let {
-                Box(
-                    modifier = Modifier.padding(top = BpkSpacing.Sm),
-                    content = it,
-                )
+                Box(modifier = Modifier.padding(top = BpkSpacing.Sm)) {
+                    Box(
+                        modifier = Modifier
+                            .size(BpkSpacing.Lg)
+                            .background(color = BpkTheme.colors.textOnDark, shape = RoundedCornerShape(BpkBorderRadius.Xs)),
```
Comment (bvitaliyg) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt:84]: The rounded corners don't work if you use `carrierLogoContent = { Box(Modifier.fillMaxSize().background(Color.Red)) }`

You need to it differently:

```
.clip(RoundedCornerShape(BpkBorderRadius.Xs))
.background(BpkTheme.colors.textOnDark)
```

**Review by bvitaliyg — COMMENTED** [review]
One small request:

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt [hunk 3]
```diff
@@ -73,13 +77,19 @@ fun BpkFlightLeg(
     ) {
         Row(modifier = Modifier.fillMaxWidth()) {
             carrierLogoContent?.let {
-                Box(
-                    modifier = Modifier.padding(top = BpkSpacing.Sm),
-                    content = it,
-                )
+                Box(modifier = Modifier.padding(top = BpkSpacing.Sm)) {
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt:80]: i think you could've just applied the padding before the size and then it treats it as a margin instead

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt (hunk 2 above)
Comment (xo0ps) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt:84]: thanks applied

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt (hunk 3 above)
Comment (xo0ps) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt:80]: yes good point, forgot about it! applied. thanks!

**Review by marianeum — APPROVED** [review]

**Review by marianeum — APPROVED** [review]

**Pull request merged**